internal/debug/debug_idl.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debug
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+	"strings"
+
+	"go.uber.org/yarpc/internal/introspection"
+)
+
+type dispatcherWithIdlTree struct {
+	Name    string
+	ID      string
+	IDLTree introspection.IDLTree
+}
+
+type idlTreeHelper struct {
+	DispatcherName string
+	Indent         int
+	Tree           *introspection.IDLTree
+}
+
+func wrapIDLTree(dname string, i int, t *introspection.IDLTree) idlTreeHelper {
+	sort.Sort(t.Modules)
+	return idlTreeHelper{dname, i + 1, t}
+}
+
+type dispatcherWithIDLModule struct {
+	Dispatcher introspection.DispatcherStatus
+	IDLModule  introspection.IDLModule
+}
+
+const idlPagePath = "/debug/yarpc/idl/"
+
+var idlPage = page{
+	path: idlPagePath,
+	handler: func(w http.ResponseWriter, req *http.Request, insp IntrospectionProvider) interface{} {
```

Comment:
you can test using idlPage.handler directly. I was trying to have some sort of name-spacing by using a struct like that. Let me know if you think of a better style to organize all the pages.